mwengine/src/main/cpp/events/baseaudioevent.cpp

```diff
@@ -101,11 +101,10 @@ void BaseAudioEvent::stop()
     }
 
     // as live events will always be rendered (meaning: also when Sequencer isn't playing),
-    // we mark the event as deletable so it can be removed from the instruments event list
-    // as soon as the engine has finished rendering the current iteration to prevent read/write conflicts
+    // we enqueue the event for removal from the instruments event list as soon as the engine
+    // has finished rendering the current iteration to prevent read/write conflicts
 
-    setDeletable( true );
-    resetPlayState();
```

Comment:
Resetting the play state would unmark an event as being a live event, meaning it could conflict with a new `play()` invocation causing a double addition within the event vector.